src/FhirClientProvider.tsx

```diff
@@ -40,64 +41,90 @@ export default function FhirClientProvider(props: Props): JSX.Element {
-                if (firstResult.resourceType === "CarePlan") {
-                    return CarePlan.from(bundle.entry[0].resource as ICarePlan);
-                } else {
-                    setError("Unexpected resource type returned");
-                    return null;
-                }
-            } else {
-                setError("Unexpected bundle type returned");
-                return null;
+    async function getCarePlans(
+      client: Client,
+      patientId: string,
+      params: object = null
+    ) {
+      if (!client) return;
+      const defaultParams = {
+        category: IsaccCarePlanCategory.isaccMessagePlan.code,
+        subject: `Patient/${patientId}`,
+        _sort: "-_lastUpdated",
+      };
+      const searchParams = new URLSearchParams({
+        ...defaultParams,
+        ...(params ?? {}),
+      }).toString();
+      return client.request(`/CarePlan?${searchParams}`).then(
+        (bundle: Bundle) => {
+          if (bundle.type === "searchset") {
+            if (!bundle.entry || !bundle.entry.length) {
+              return null;
+            }
+            if (bundle.total > 1) {
+              console.log(
+                "Multiple ISACC CarePlans found. Using the most recently updated.",
+                bundle
+              );
             }
-        }, (reason: any) => {
-            setError(reason.toString());
+            return bundle.entry.map((item) =>
+              CarePlan.from(item.resource as ICarePlan)
+            );
+          } else {
+            setError("Unexpected bundle type returned");
             return null;
-        });
+          }
+        },
+        (reason: any) => {
+          setError(reason.toString());
+          return null;
+        }
+      );
     }
 
-
-
     React.useEffect(() => {
         FHIR.oauth2.ready().then(
             (client: Client) => {
                 setClient(client);
-                getPatient(client).then((patientResult: Patient) => {
+                getPatient(client)
+                  .then((patientResult: Patient) => {
                     setPatient(patientResult);
                     console.log(`Loaded ${patientResult.reference}`);
-                    getCarePlan(client, patientResult.id).then((carePlanResult: CarePlan) => {
-                        setCarePlan(carePlanResult);
-                        if (carePlanResult) {
-                            console.log(`Loaded ${carePlanResult.reference}`);
-                        }
-                        setLoaded(true);
-                    }, (reason: any) => setError(reason)).catch(e => {
-                        console.log("Error fetching CarePlan", e)
+                    getCarePlans(client, patientResult.id)
+                      .then(
+                        (carePlanResults: CarePlan[]) => {
+                          const activeCarePlans = carePlanResults?.filter(
+                            (item) => item.status === "active"
```

Comment:
@achen2401  I *think* we want all CarePlans, regardless of their `status`. But, I'm not certain, as I don't know how we manage `status` for CP's.

Reply:
thanks @mcjustin noting that, yes, the messages for all CarePlans will be displayed regardless of status, see [here](https://github.com/uwcirg/isacc-messaging-client-sof/pull/34/files#diff-ecc3420995ac53a698706c9b76aea2e203c950f91b7526d190907e893e27ae0dR132) 
The change is to get the current care plan (status == active) and also [all the careplans](https://github.com/uwcirg/isacc-messaging-client-sof/pull/34/files#diff-f985d4d496ba46e6fadc4d6a1531877caeaa241f9eccf6a885e4e6839ce8ea98R109) for the patient.